libminifi/include/utils/file/FileUtils.h

```diff
@@ -171,6 +179,19 @@ inline const std::optional<std::filesystem::file_time_type> last_write_time(cons
   return std::nullopt;
 }
 
+inline std::optional<std::string> get_last_modified_time_formatted_string(const std::string& path, const std::string& format_string) {
+  auto last_write = last_write_time(path);
+  if (!last_write) {
+    return std::nullopt;
+  }
+  auto last_write_time_t = to_time_t(*last_write);
+  std::array<char, 128U> result;
+  if (std::strftime(result.data(), result.size(), format_string.c_str(), gmtime(&last_write_time_t)) != 0) {
+    return std::string(result.data());
+  }
+  return std::nullopt;
```

Comment:
Looks good, but unfortunately it's not easy to implement. The problem is that the file_time_type specific functions are still defined in the FileUtils like to_time_t which is used by this function so I cannot move it to TimeUtils without the need to move those functions as well to TimeUtils. The other problem is that `format_time` would also return an `std::optional` which would result in a `std::optional<std::optional<std::string>>` return value.

Reply:
For the second problem, you can use `utils::flatMap` instead.

Reply:
I didn't realize this was a `file_time_type`, not a `time_point<system_clock>`.  Since this is file-related, it's OK to keep it in FileUtils; but I would still move the time formatting code to a separate `format_file_time()` function.